### ISSUE TITLE
nanoKVM doesnt work with Mac mini

### ISSUE THREAD

**Issue by haggis-supper** [sipeed/NanoKVM#119]
I've plugged both a NanoKVM lite and a NanoKVM full into a MAC mini, they both show the screen but the mouse and keyboard dont work.

**Comment by MetalSamurai99** [comment]
I have 3 NanoKVM (full) attached to Mac minis. 

For 1 running MacOS 10.12.6 the screen updates correctly, the keyboard works, but the mouse movements are very unpredictable and basically useless

The other two are running XCP-NG (Linux, I think kernel 5.x) with text console and they work just fine. 

I did test with an old Raspberry Pi 1B and again video capture works, but in that case I never got keyboard input working at all. 

All are set up with POE splitters for power, not drawing power from the host USB port.

**Comment by haggis-supper** [comment]
Yep as you said, i'm running Mac Os 12.7.6, like youve said the screen updates but i dont get any life from Keyboard or Mouse. 

Linux hosts work fine, I will check my Mac mini with my PiKVM ( and see how that goes )

**Comment by pfrench42** [comment]
Mine just arrived today.
Hooked it up to a 2012 Mac Mini.
Keyboard and Mouse worked , even through a reboot of the Mini.
After about 20 minutes, keyboard and mouse are unresponsive on the mini
I rebooted the NanoKVM and key/mouse came back up on the Mini

**Comment by jasonhouston** [comment]
I am noticing that mine intermittently stop responding to keyboard/mouse on Linux hosts, a reboot generally fixes it but I have a couple that are persistently failing to respond to keyboard/mouse control.

**Comment by haggis-supper** [comment]
New code now works with Mac minis and seems to be very stable.

**Comment by vipavlovic** [comment]
Mine works with my M3 Macbook Pro, but it has problems with CMD+anything.  E.g., both CMD+C and CMD+V result in an infinite repetition of the same command, until I press ESC.  Any way to fix this?

**Comment by tjcomserv** [comment]
How are you accessing the KVM? Via a Mac device? Using Safari/Chrome?

**Comment by vipavlovic** [comment]
> How are you accessing the KVM? Via a Mac device? Using Safari/Chrome?

Yes, via a Mac device.  Tried several browsers, including Safari and Chrome, both result in the same behavior - repeated keystrokes.

**Comment by tjcomserv** [comment]
I have just had a similar issue that I resolved by ensuring that the HID USB cable AND the power USB cable were both connected to the PC/Mac I wanted to control. If KVM is powered by another PC or wall plug then neither the keyboard and mouse work.

**Comment by Myke** [comment]
HID-Only mode seems to do the trick on the MacMini I've been testing with.
